### DISCUSSION TITLE
How to add custom trusted_hosts when working with django+nginx+proxy

### DISCUSSION THREAD

**Discussion by jeromelefeuvre** [pallets/werkzeug#2917] [Q&A] [answered]
Hi,

I would like to implement interactive debugger on my django app. 

I configured a local domain where nginx has a proxy  to my django container. Let's says `dev.foo.com`.
```
location / {
                set $upstream http://foo:8000;
                proxy_pass $upstream;
                proxy_redirect off;
                proxy_set_header Host $host;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header X-Forwarded-Proto $scheme;
                proxy_read_timeout 300;                
}
```

I saw that werkzeug test if my domain is on the trusted_hosts but the listed datas are just local ips:
https://github.com/pallets/werkzeug/blob/62e3ea45846d06576199a2f8470be7fe44c867c1/src/werkzeug/debug/__init__.py#L303

My question is how to add my local domain to the trusted_hosts. I saw some exemple for flasks but not for django...

Thanks!

**Comment by davidism** [comment, marked as answer]
Most operating systems are now pre-configured to treat `*.localhost` as `127.0.0.1`. So my strategy for using domains during local development is `project.localhost`, `api.localhost`, `auth.api.localhost` etc. If you use that, everything will already work.

Otherwise, the debugger is implemented as WSGI middleware. I'm not entirely sure how to add WSGI middleware to Django (as opposed to Django middelware), but it's definitely possible.

```python
app = django.get_wsgi_app()  # something like this
debug_app = DebuggedApplication(app, evalex=True)
debug_app.trusted_hosts = [...]   # your own hosts here, or from django config
# somehow tell the django dev server to run this app instead
```

**Comment by jeromelefeuvre** [comment]
Perfect! that's exactly that! I was close to the solution. On my side, it's more:
````
from django.core.wsgi import get_wsgi_application
from werkzeug.debug import DebuggedApplication

application = get_wsgi_application()

application = DebuggedApplication(application, evalex=True)
application.trusted_hosts = ['foo.nfb.ca']
````

Thanks !
